php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1100 @@
+    );
+  }
+
+  public ForStatementTree forStatement(ForStatementHeader forStatementHeader, StatementTree statement) {
+    return new ForStatementTreeImpl(forStatementHeader, statement);
+  }
+
+  public ForStatementTree forStatementAlternative(
+      ForStatementHeader forStatementHeader, InternalSyntaxToken colonToken,
+      Optional<List<StatementTree>> statements, InternalSyntaxToken endForToken, InternalSyntaxToken eos
+  ) {
+    return new ForStatementTreeImpl(forStatementHeader, colonToken, optionalList(statements), endForToken, eos);
+  }
+
+  public SeparatedList<ExpressionTree> forExpr(ExpressionTree expression, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> listOptional) {
+    return separatedList(expression, listOptional);
+  }
+
+  public ElseClauseTree elseClause(InternalSyntaxToken elseToken, StatementTree statement) {
+    return new ElseClauseTreeImpl(elseToken, statement);
+  }
+
+  public IfStatementTree ifStatement(
+      InternalSyntaxToken ifToken, ParenthesisedExpressionTree expression, StatementTree statement,
+      Optional<List<ElseifClauseTree>> elseIfClauses, Optional<ElseClauseTree> elseClause
+  ) {
+    return new IfStatementTreeImpl(ifToken, expression, statement, optionalList(elseIfClauses), elseClause.orNull());
+  }
+
+  public ElseifClauseTree elseifClause(InternalSyntaxToken elseifToken, ParenthesisedExpressionTree condition, StatementTree statement) {
+    return new ElseifClauseTreeImpl(elseifToken, condition, statement);
+  }
+
+  public IfStatementTree alternativeIfStatement(
+      InternalSyntaxToken ifToken, ParenthesisedExpressionTree condition, InternalSyntaxToken colonToken,
+      Optional<List<StatementTree>> statements, Optional<List<ElseifClauseTree>> elseifClauses, Optional<ElseClauseTree> elseClause,
+      InternalSyntaxToken endIfToken, InternalSyntaxToken eosToken
+  ) {
+    return new IfStatementTreeImpl(
+        ifToken,
+        condition,
+        colonToken,
+        optionalList(statements),
+        optionalList(elseifClauses),
+        elseClause.orNull(),
+        endIfToken,
+        eosToken
+    );
+  }
+
+  public ElseClauseTree alternativeElseClause(InternalSyntaxToken elseToken, InternalSyntaxToken colonToken, Optional<List<StatementTree>> statements) {
+    return new ElseClauseTreeImpl(
+        elseToken,
+        colonToken,
+        optionalList(statements)
+    );
+  }
+
+  public ElseifClauseTree alternativeElseifClause(
+      InternalSyntaxToken elseifToken, ParenthesisedExpressionTree condition, InternalSyntaxToken colonToken,
+      Optional<List<StatementTree>> statements
+  ) {
+    return new ElseifClauseTreeImpl(
+        elseifToken,
+        condition,
+        colonToken,
+        optionalList(statements)
+    );
+  }
+
+  public DoWhileStatementTree doWhileStatement(InternalSyntaxToken doToken, StatementTree statement, InternalSyntaxToken whileToken, ParenthesisedExpressionTree condition, InternalSyntaxToken eosToken) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 203 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)